tcms/rpc/tests/test_testrun.py

```diff
@@ -251,6 +301,48 @@ def test_create_with_product_version(self):
         self.assertEqual(result['product_version'], self.plan.product_version.value)
 
 
+class TestFilter(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.plan = TestPlanFactory()
+
+        self.test_case = TestCaseFactory()
+        self.test_case.save()
+        self.plan.add_case(self.test_case)
+
+        self.test_run = TestRunFactory(plan=self.plan)
+
+    def test_empty_query(self):
+        result = self.rpc_client.TestRun.filter()
+        self.assertTrue(isinstance(result, list))
+        self.assertEqual(1, len(result))
+        self.assertEqual(self.test_run.pk, result[0]['id'])
+
+    def test_filter(self):
+        result = self.rpc_client.TestRun.filter({'plan': self.plan.pk, 'stop_date__isnull': True})
```

Comment:
stop_date is redundant a bit. But if you really want to make sure that the filtering works as expected there should be a second test run which belongs to another plan. Hint: if you don't specify it as parameter to the factory it will be created automatically.

Reply:
thanks for the note. that is what i want to do.